tests/fixtures/optimizer/simplify.sql

```diff
@@ -911,13 +911,58 @@ t1.a = 39 AND t2.b = t1.a AND t3.c = t2.b;
 t1.a = 39 AND t2.b = 39 AND t3.c = 39;
 
 x = 1 AND CASE WHEN x = 5 THEN FALSE ELSE TRUE END;
-x = 1 AND CASE WHEN FALSE THEN FALSE ELSE TRUE END;
+x = 1;
 
 x = 1 AND IF(x = 5, FALSE, TRUE);
-x = 1 AND CASE WHEN FALSE THEN FALSE ELSE TRUE END;
+x = 1;
+
+-- Simplified to x = 1 if the canonicalize rule runs first
+x = 1 AND CASE x WHEN 5 THEN FALSE ELSE TRUE END;
+x = 1 AND CASE 1 WHEN 5 THEN FALSE ELSE TRUE END;
 
 x = y AND CASE WHEN x = 5 THEN FALSE ELSE TRUE END;
 x = y AND CASE WHEN x = 5 THEN FALSE ELSE TRUE END;
 
 x = 1 AND CASE WHEN y = 5 THEN x = z END;
 x = 1 AND CASE WHEN y = 5 THEN 1 = z END;
+
+--------------------------------------
+-- Simplify Conditionals
+--------------------------------------
+IF(TRUE, x, y);
+x;
+
+IF(FALSE, x, y);
+y;
+
+IF(FALSE, x);
+NULL;
+
+IF(NULL, x, y);
+CASE WHEN NULL THEN x ELSE y END;
+
+IF(cond, x, y);
+CASE WHEN cond THEN x ELSE y END;
+
+CASE WHEN TRUE THEN x ELSE y END;
+x;
+
+CASE WHEN FALSE THEN x ELSE y END;
+y;
+
+CASE WHEN FALSE THEN x WHEN FALSE THEN y WHEN TRUE THEN z END;
+z;
+
+-- Simplified to w if the canonicalize rule runs first
```

Comment:
canonicalization never runs before simplify, i think we should get rid of the canonicalizer

Reply:
Yeah not sure if there's much value to keeping them separated besides making the intent of these transformations more clear (as in: one "canonicalizes", the other actually "simplifies"). Another thing could be that canonicalizations require type inference but that could be easily checked before calling them to avoid that overhead.

Reply:
Let's do this in a followup PR, cc @barakalon

Reply:
I guess canonicalize and simplify have become a dumping grounds for rules that operate on simple expressions, while most of the other rules tend to deal with SELECT.

I think merging them makes sense.

Reply:
sorry, i didn't mean get rid of canonicalizer, i meant move this rule from canocialize to simplify

Reply:
Ah 😁

Is there a value to keeping the canonicalize rule around though? I was also thinking what Barak said.

Reply:
because you need types

Reply:
we can have a section in the simplifier where we check if there are types, and if so then apply the functions in canonicalize.py that need them (there are a couple that don't even need types)